pykusto/_src/query.py

```diff
@@ -545,8 +545,32 @@ def __init__(self, head: Query, joined_query: Query, kind: JoinKind,
         self._kind = kind
         self._on_attributes = on_attributes
 
-    def on(self, col1: _AnyTypeColumn, col2: _AnyTypeColumn = None) -> '_JoinQuery':
-        self._on_attributes = self._on_attributes + (((col1,),) if col2 is None else ((col1, col2),))
+    def on(self, join_cols: Union[_AnyTypeColumn, Tuple[_AnyTypeColumn, ...], List[Union[_AnyTypeColumn, Tuple[_AnyTypeColumn, ...]]]]) -> '_JoinQuery':
```

Comment:
As we said f2f, we still allow it, but with a slight exception.
The exception is when joining using a different column name in each of the tables (the left.col, right.col case).
In such a case, a tuple is required and not explicit two arguments, otherwise, it will evaluate as two different join columns.
This might be not backward compatible.